src/shared/GitHub/GitHubAuthentication.cs

```diff
@@ -79,6 +76,19 @@ public async Task<AuthenticationPromptResult> GetAuthenticationAsync(Uri targetU
                 throw new ArgumentException(@$"Must specify at least one {nameof(AuthenticationModes)}", nameof(modes));
             }
 
+            switch (modes)
+            {
+                case AuthenticationModes.Browser:
+                    // no user interaction required
+                    return new AuthenticationPromptResult(AuthenticationModes.Browser);
+                case AuthenticationModes.Device:
+                    // no user interaction required
+                    return new AuthenticationPromptResult(AuthenticationModes.Device);
+                default:
+                    break;
+            }
```

Comment:
Let's be consistent with the above `if (modes == .None)` style here:

```suggestion
            // If there is no mode choice to be made and no interaction required,
            // just return that result.
            if (modes == AuthenticationModes.Browser ||
                modes == AuthenticationModes.Device)
            {
                    return new AuthenticationPromptResult(modes);
            }
```